modules/http/src/test/java/org/mule/module/http/api/HttpListenerBuilderTestCase.java

```diff
@@ -7,52 +7,43 @@
 package org.mule.module.http.api;
 
 import static org.junit.Assert.assertThat;
-import static org.mockito.Matchers.any;
 import static org.mockito.Matchers.anyString;
-import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
-
-import org.mule.api.MuleContext;
-import org.mule.api.context.WorkManagerSource;
 import org.mule.construct.Flow;
-import org.mule.execution.MessageProcessingManager;
 import org.mule.module.http.api.listener.HttpListener;
 import org.mule.module.http.api.listener.HttpListenerBuilder;
 import org.mule.module.http.api.listener.HttpListenerConfig;
 import org.mule.module.http.internal.listener.DefaultHttpListenerConfig;
 import org.mule.module.http.internal.listener.HttpListenerConnectionManager;
 import org.mule.module.http.internal.listener.ServerAddress;
-import org.mule.tck.junit4.AbstractMuleTestCase;
-import org.mule.tck.size.SmallTest;
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.tck.junit4.rule.DynamicPort;
 import org.mule.transport.ssl.api.TlsContextFactory;
 
 import java.net.MalformedURLException;
 import java.net.URL;
-import java.util.Arrays;
-import java.util.Collections;
 
 import org.hamcrest.core.Is;
+import org.junit.Rule;
 import org.junit.Test;
-import org.mockito.Answers;
 
-@SmallTest
-public class HttpListenerBuilderTestCase extends AbstractMuleTestCase
+public class HttpListenerBuilderTestCase extends FunctionalTestCase
```

Comment:
this is needed for spring to detect dependencies and apply correct initialisation order